Add `dates_are_gmt` parameter to REST API

TimBHowe opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Added the `dates_are_gmt` parameters to allow the before and after parameters to use the `post_date_gmt` column if it is true.

Closes #28111 

**Related PR in documentation repository:** https://github.com/woocommerce/woocommerce-rest-api-docs/pull/203

### How to test the changes in this Pull Request:

1. Install WordPress/WooCommerce with testing data
2. Generate some API Keys
3. Make a REST API request to the 'https://localhost.com/wp-json/wc/v2/orders' end point and use the dates_are_gmt parameters to allow the before and after parameters to use the post_date_gmt column if it is true.
Example: https://localhost.com/wp-json/wc/v3/orders?dates_are_gmt=true&after=2021-04-24T06:00:00&before=2021-04-24T14:00:00&consumer_key=xxx&consumer_secret=xxx

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Added the `dates_are_gmt` parameters to REST API to searched posts using the `post_date_gmt` column.

TimBHowe committed:
Added the 'after_gmt' and 'before_gm' parameters

Konamiman commented:
Hi @TimBHowe, thanks for your contribution. The problem I see with your approach is that it makes `before/after` and `before/after_gmt` arguments mutually exclusive, if one is applied then the other will necessarily be ignored and which one will "win" is accidental based on the order of their processing in the code.

What if instead you add a new bool `dates_are_gmt` argument? So that setting it to `true` will cause the existing `before` and `after` arguments to be interpreted as gmt dates.

Additionally, any addition or change in request arguments needs to be accompanied by the corresponding update in the `get_collection_params` method in the same class.

TimBHowe committed:
Added dates_are_gmt parameter to allow the uses of post_date_gmt column with the before and after parameters for the order RESTAPI

TimBHowe commented:
Updated the pull requests as requested.

Added the `dates_are_gmt` parameters to allow the `before` and `after` parameters to use the post_date_gmt column if it is true.

Also added the new parameter to `get_collection_params` function in the class.

The new example request would look something like this
https://localhost.com/wp-json/wc/v3/orders?dates_are_gmt=true&after=2021-04-24T06:00:00&before=2021-04-24T14:00:00&consumer_key=xxx&consumer_secret=xxx

Konamiman reviewed (approved):
Looks good, I'm approving. Could you please update the pull request title and description? They're still referring to the old `before_gmt` and `after_gmt` parameters. I'll merge afterwards.

TimBHowe commented:
Updated the description of the pull request.

Let me know if you need any other changes.

Konamiman commented:
It's all good now, merging.
